src/zenml/integrations/vertex/orchestrators/vertex_orchestrator.py

```diff
@@ -0,0 +1,373 @@
+                    f"""
+                    name: {step.name}
+                    implementation:
+                        container:
+                            image: {image_name}
+                            command: {command + arguments}"""
+                )()
+
+                # Set upstream tasks as a dependency of the current step
+                upstream_step_names = self.get_upstream_step_names(
+                    step=step, pb2_pipeline=pb2_pipeline
+                )
+                for upstream_step_name in upstream_step_names:
+                    upstream_container_op = step_name_to_container_op[
+                        upstream_step_name
+                    ]
+                    container_op.after(upstream_container_op)
+
+                step_name_to_container_op[step.name] = container_op
+
+        # Save the generated pipeline to a file.
+        assert runtime_configuration.run_name
+        fileio.makedirs(self.pipeline_directory)
+        pipeline_file_path = os.path.join(
+            self.pipeline_directory,
+            f"{runtime_configuration.run_name}.json",
+        )
+
+        # Compile the pipeline using the Kubeflow SDK V2 compiler that allows
+        # to generate a JSON representation of the pipeline that can be later
+        # upload to Vertex AI Pipelines service.
+        logger.debug(
+            "Compiling pipeline using Kubeflow SDK V2 compiler and saving it to %s",
+            pipeline_file_path,
+        )
+        KFPV2Compiler().compile(
+            pipeline_func=_construct_kfp_pipeline,
+            package_path=pipeline_file_path,
+            pipeline_name=_clean_pipeline_name(pipeline.name),
+        )
+
+        # Using the Google Cloud AIPlatform client, upload and execute the pipeline
+        # on the Vertex AI Pipelines service.
+        self._upload_and_run_pipeline(
+            pipeline_name=pipeline.name,
+            pipeline_file_path=pipeline_file_path,
+            runtime_configuration=runtime_configuration,
+            enable_cache=pipeline.enable_cache,
+        )
+
+    def _upload_and_run_pipeline(
+        self,
+        pipeline_name: str,
+        pipeline_file_path: str,
+        runtime_configuration: "RuntimeConfiguration",
+        enable_cache: bool,
+    ) -> None:
+        """Uploads and run the pipeline on the Vertex AI Pipelines service.
+
+        Args:
+            pipeline_name: Name of the pipeline.
+            pipeline_file_path: Path of the JSON file containing the compiled
+                Kubeflow pipeline (compiled with Kubeflow SDK v2).
+            runtime_configuration: Runtime configuration of the pipeline run.
+            enable_cache: Whether caching is enabled for this pipeline run.
+        """
+
+        # We have to replace the hyphens in the pipeline name with underscores
+        # and lower case the string, because the Vertex AI Pipelines service
+        # requires this format.
+        assert runtime_configuration.run_name
```

Comment:
Would be nice if we could add a warning that schedules are ignored/not possible with vertex ai pipelines.
```python
if runtime_configuration.schedule:
    logger.warning("...")
```

Reply:
Agree 100%. Done!